OpenRA.Mods.Common/HitShapes/IHitShape.cs

```diff
@@ -0,0 +1,21 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public interface IHitShape
```

Comment:
tl;dr: ```WDist DistanceFromEdge(WVec p);``` might already suit the needs of #9357.

!tl;dr
With regards to #9357, does the implementations of the line below ```WDist DistanceFromEdge(WVec p);``` give you the distance from the actor center position to the edge of its shape along a vector?

If so, in #9357 the [actor width line] (https://github.com/OpenRA/OpenRA/pull/9357/files#diff-5f66c498eea5ea840b8d556a7285788eR49) can be adjusted to use this instead of assuming circle radius (i.e. equal radius in all directions).

Basically, I could use it to get the actor's 'width' along the vector from its center point to the closest point on the line.